cmd/query/app/server.go

```diff
@@ -274,7 +274,7 @@ func (s *Server) initListener() (cmux.CMux, error) {
 func (s *Server) Start() error {
```

Comment:
you should add this option to the logger in the test:
```
	flagsSvc.Logger = zaptest.NewLogger(t, zaptest.WrapOptions(zap.AddCaller()))
```
that way it prints the source line.

I changed the timeout to 2s and look what the logs say:
```
    logger.go:146: 2024-06-12T15:13:20.232-0400 INFO    app/static_handler.go:109       Using UI configuration  {"path": ""}
    logger.go:146: 2024-06-12T15:13:20.234-0400 INFO    app/server.go:256       Query server started    {"port": 16686, "addr": ":16686"}
    logger.go:146: 2024-06-12T15:13:20.234-0400 INFO    app/server.go:300       Starting HTTP server    {"port": 16686, "addr": ":16686"}
    logger.go:146: 2024-06-12T15:13:20.234-0400 INFO    app/server.go:332       Starting CMUX server    {"port": 16686, "addr": ":16686"}
    logger.go:146: 2024-06-12T15:13:20.234-0400 INFO    app/server.go:318       Starting GRPC server    {"port": 16686, "addr": ":16686"}
    logger.go:146: 2024-06-12T15:13:20.283-0400 INFO    app/server.go:358       Stopping gRPC server
    logger.go:146: 2024-06-12T15:13:20.284-0400 INFO    app/server.go:310       HTTP server stopped     {"port": 16686, "addr": ":16686"}
    logger.go:146: 2024-06-12T15:13:20.284-0400 INFO    app/server.go:323       GRPC server stopped     {"port": 16686, "addr": ":16686"}
    logger.go:146: 2024-06-12T15:13:22.285-0400 INFO    app/server.go:362       Closing HTTP server
    logger.go:146: 2024-06-12T15:13:24.286-0400 INFO    app/server.go:369       Closing CMux server
    logger.go:146: 2024-06-12T15:13:24.286-0400 INFO    app/server.go:375       Server stopped
```
note that after `Stopping gRPC server` we're seeing `HTTP server stopped`, even before `Closing HTTP server`.  When I add a log line for CMUX server goroutine, it also logs that CMUX server is stopped before we come to trying to stop HTTP. So it seems when we're running on a single port it's simply sufficient to close gRPC server, which closes the underlying listenter, and all other servers exit automatically (we do already catch `ErrServerClosed` errors and don't log them).